mkdocs/contrib/search/search_index.py

```diff
@@ -216,7 +225,7 @@ def handle_data(self, data: str) -> None:
         # Otherwise it is content of something under that header
         # section.
         if self.is_header_tag:
-            self.section.title = data
+            self.section.title = self.section.title + data
```

Comment:
What is this "data" and why is it OK to concatenate these two strings? How do they each look that it's OK to concatenate them without even a space in between? Can you add a code comment?

Reply:
This is the Python Standard lib HTMLParser. Data is anything that is not part of a tag. So it includes all text and whitespace. For example, consider this HTML:

``` HTML
<p>
foo <span>bar</span> baz
</p>
```
This method would be called 3 times and in each respective call, `data` would contain the following values:

1. `\nfoo `
2. `bar`
3. ` baz\n`

Concatenate those together, and you get the plaintext content of the `p` element: `\nfoo bar baz\n`. Notice that whitespace is preserved. There is no need to add any.

Not sure why we need a code comment. This is standard use here.

Reply:
By the way, this is the bug I mentioned at the end of my first comment. Previously, each call to the function would have overwritten the previous value so that the final value would be only ` baz\n`. This fixes that bug.